Simplenote/src/main/java/com/automattic/simplenote/repositories/SimperiumCollaboratorsRepository.kt

```diff
@@ -14,4 +31,83 @@ class SimperiumCollaboratorsRepository @Inject constructor() : CollaboratorsRepo
     override fun isValidCollaborator(collaborator: String): Boolean {
         return !TextUtils.isEmpty(collaborator) && EMAIL_ADDRESS.matcher(collaborator).matches()
     }
+
+    /**
+     * Return a list of collaborators (email addresses as tags) if the note for the given simperiumKey ([noteId]) is
+     * not in the trash and has not been deleted
+     */
+    override suspend fun getCollaborators(noteId: String): CollaboratorsActionResult = withContext(ioDispatcher) {
+        return@withContext when(val result = getNote(noteId)) {
+            is Either.Left -> result.l
+            is Either.Right ->
+                CollaboratorsActionResult.CollaboratorsList(filterCollaborators(result.r))
+        }
+    }
+
+    override suspend fun addCollaborator(noteId: String, collaborator: String): CollaboratorsActionResult =
+        withContext(ioDispatcher) {
+            return@withContext when (val result = getNote(noteId)) {
+                is Either.Left -> result.l
+                is Either.Right -> {
+                    val note = result.r
+                    note.addTag(collaborator)
+                    CollaboratorsActionResult.CollaboratorsList(filterCollaborators(note))
+                }
+            }
+        }
+
+    override suspend fun removeCollaborator(noteId: String, collaborator: String): CollaboratorsActionResult =
+        withContext(ioDispatcher) {
+            return@withContext when(val result = getNote(noteId)) {
+                is Either.Left -> result.l
+                is Either.Right -> {
+                    val note = result.r
+                    note.removeTag(collaborator)
+                    CollaboratorsActionResult.CollaboratorsList(filterCollaborators(note))
+                }
+            }
+        }
+
+    override suspend fun collaboratorsChanged(noteId: String): Flow<Boolean> = callbackFlow {
```

Comment:
Fixed in 56de650bed91f02c939abdcf2ce662415da8341d. I had to mock `CollaboratorsRepository` in order to be able to test this function. Other approaches such as `spy` the class or partial mocks were not straightforward.

Reply:
👋 @danilo04 !

Thanks for adding those two test. However, I am a bit confused by the value added here, can we discuss it a bit before merging? 🤔 

Question (❓): In addition the test you added on the `ViewModel` layer, is it possible to add `Repository` layer test for the `collaboratorsChanged()`? PS: Same question applies to `SimperiumTagsRepository.tagsChanged()`.

Reply:
Hey @ParaskP7 👋 !

I need to find a way to inject the calls to notify the callbacks. Since we are using mocks for the `Bucket` objects, I need to figure out whether a partial mock or some other technique helps me to be able to trigger the different callbacks. Do you have any idea how we could test these flows at the repository level?

Reply:
👋 @danilo04 !

> Do you have any idea how we could test these flows at the repository level?
Yeap, I am not sure, I would need to think and play a bit with it before me suggesting a way to add these tests. Please feel free to progress without that change, but can you please add an small issue on that so that we can revisit it later. 🙏